src/mango/test/12-use-correct-index-test.py

```diff
@@ -54,36 +54,41 @@ def setUp(self):
         self.db.save_docs(copy.deepcopy(DOCS))
 
     def test_choose_index_with_one_field_in_index(self):
-        self.db.create_index(["name", "age", "user_id"], ddoc="aaa")
-        self.db.create_index(["name"], ddoc="zzz")
+        self.db.create_index(["name", "age", "user_id"], ddoc="aaa", wait_for_built_index=False)
+        self.db.create_index(["name"], ddoc="zzz", wait_for_built_index=False)
+        self.db.wait_for_built_indexes()
```

Comment:
This `wait_for_built_indexes` looks like a major change in API behavior. I'm assume this is because of the filter for indexes that are currently building which leads to no valid index? I'd wager that users would generally not enjoy that sudden change. Perhaps if we get the no valid index we instead go back and wait for the build rather than throwing an error? Users that are savvy and want to wait can do so with the index info.

Reply:
This is a tricky one. Because if we have large indexes and a user creates a new index and then runs a query that would use that index they could wait a long time for that index to finish building versus choosing another index that is built but isn't quite as good a fit.

Reply:
What's tricky about it? You can do the check to only consider built indices which covers your case, then if nothing is built, go back to waiting for one. I guess theoretically there might be fun games to play around for "wait for the first one of these" but that also seems like a future optimization.

Reply:
Based on the dev@ discussion it seems like we're free to implement waiting for an index to build if one would service a selector. Doing so would remove a large amount of code churn here and maintain the behavior previous to this PR while also preventing the operational issues around creating an index that slows down existing queries.